File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.rb
# @param {String} s
# @return {Integer}
def max_depth(s)
    result=[]
    count=0
    maxcount=0
    s.chars.each_index do |i|
        if s[i] == '('
            count+=1
        elsif s[i] ==')'
            count-=1
        end
        maxcount = count if count > maxcount
    end
   maxcount
end

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<char> stack;
        int maxcount=0;
        for (const auto& chr : s) {
            if ( chr == '(' ) stack.push(chr);
            if ( chr == ')' ) stack.pop();
            if (stack.size() > maxcount) maxcount = stack.size();
        }
        return maxcount;
    }
};